test/test_batterystate.py

```diff
@@ -0,0 +1,282 @@
+        msg.percentage = 0.
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)  # let callbacks finish
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.ACTIVE)
+
+    def test_discharging(self):
+        msg = BatteryState()
+        # Discharging at max, report FAIL
+        msg.power_supply_status = BatteryState.POWER_SUPPLY_STATUS_DISCHARGING
+        msg.percentage = 1.
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.FAIL)
+
+        # Discharging but above max, report FAIL
+        self.blackboard.save("max_charge", 0.95, self.charge_monitor._id)
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.FAIL)
+
+        # Discharging but below max, report FAIL
+        msg.percentage = 0.
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)  # let callbacks finish
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.FAIL)
+
+    def test_full(self):
+        msg = BatteryState()
+        # Charge at full, report SUCCESS.
+        msg.power_supply_status = BatteryState.POWER_SUPPLY_STATUS_FULL
+        msg.percentage = 1.
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.SUCCESS)
+
+        # Charge at full but above max, report SUCCESS
+        self.blackboard.save("max_charge", 0.95, self.charge_monitor._id)
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.SUCCESS)
+
+        # Charge at full, but percentage low, still report success
+        msg.percentage = 0.
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)  # let callbacks finish
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.SUCCESS)
+
+    def test_unknown(self):
+        msg = BatteryState()
+        # Unkown at max, report FAIL
+        msg.power_supply_status = BatteryState.POWER_SUPPLY_STATUS_UNKNOWN
+        msg.percentage = 1.
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.FAIL)
+
+        # Unkown but above max, report FAIL
+        self.blackboard.save("max_charge", 0.95, self.charge_monitor._id)
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.FAIL)
+
+        # Unknown but below max, report FAIL
+        msg.percentage = 0.
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)  # let callbacks finish
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.FAIL)
```

Comment:
I worry a bit about declaring unknown values as success or fail. perhaps add a line to the class doc specifying this behavior so callers can plan accordingly?